meteor/client/ui/RundownView.tsx

```diff
@@ -2009,39 +2010,54 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
 
 		renderSegments() {
 			if (this.props.segments) {
-				return this.props.segments.map((segment, index, array) => {
-					if (this.props.studio && this.props.playlist && this.props.showStyleBase) {
-						return (
-							<ErrorBoundary key={unprotectString(segment._id)}>
-								<VirtualElement
-									id={SEGMENT_TIMELINE_ELEMENT_ID + segment._id}
-									margin={'100% 0px 100% 0px'}
-									initialShow={index < window.innerHeight / 260}
-									placeholderHeight={260}
-									placeholderClassName="placeholder-shimmer-element segment-timeline-placeholder"
-									width="auto">
-									<SegmentTimelineContainer
-										id={SEGMENT_TIMELINE_ELEMENT_ID + segment._id}
-										studio={this.props.studio}
-										showStyleBase={this.props.showStyleBase}
-										followLiveSegments={this.state.followLiveSegments}
-										segmentId={segment._id}
-										playlist={this.props.playlist}
-										liveLineHistorySize={this.LIVELINE_HISTORY_SIZE}
-										timeScale={this.state.timeScale}
-										onTimeScaleChange={this.onTimeScaleChange}
-										onContextMenu={this.onContextMenu}
-										onSegmentScroll={this.onSegmentScroll}
-										isLastSegment={index === array.length - 1}
-										onPieceClick={this.onSelectPiece}
-										onPieceDoubleClick={this.onPieceDoubleClick}
-										onHeaderNoteClick={(level) => this.onHeaderNoteClick(segment._id, level)}
-									/>
-								</VirtualElement>
-							</ErrorBoundary>
-						)
+				console.log('this.props.segments', this.props.segments)
+
+				let elements: (JSX.Element | undefined)[] = []
+
+				let previousRundownId = ''
+				this.props.rundowns.forEach((rundown, index) => {
+					if (this.props.rundowns.length > 1) {
+						elements.push(<RundownDividerTimeline key={`rundown${index}`} rundown={rundown} />)
 					}
+					elements = elements.concat(
+						this.props.segments
+							.filter((segment) => segment.rundownId === rundown._id)
```

Comment:
This will perform very poorly and result in a long loading time and bad performance on takes. The resulting operations will be:
1) Scan the Segments collection and select segments matching the rundowns in the playlist and order them correctly. This is a pretty costly operation.
2) For every rundown, scan the entire array resulting from step 1. and find the rundowns matching the current rundown in the playlist. Note that this will run every time there is a significant change in the playlist or a segment is added.